File: iRead/iReader/App/AppDelegate+Debug.m
//
//  AppDelegate+Debug.m
//  iReader
//
//  Created by zzyong on 2018/10/8.
//  Copyright © 2018年 zouzhiyong. All rights reserved.
//

#import "AppDelegate+Debug.h"
#import <FLEX.h>
#import "IRDebugConst.h"
#import "DTCoreTextLayoutFrame.h"

static UIWindow *flexWindow;

@implementation AppDelegate (Debug)

- (void)debugInit
{
    [self addFLEX];
    
    BOOL dtEndble = [[[IRCacheManager sharedInstance] objectForKey:kDTCoreTextDebugEnableCacheKey] boolValue];
    [DTCoreTextLayoutFrame setShouldDrawDebugFrames:dtEndble];
}

- (UIWindow *)flexWindow
{
    return flexWindow;
}

- (void)addFLEX
{
    flexWindow = [[UIWindow alloc] init];
    flexWindow.backgroundColor = [UIColor clearColor];
    flexWindow.rootViewController = [[UIViewController alloc] init];
    flexWindow.windowLevel = UIWindowLevelStatusBar + 50;
    CGFloat windowY = CGRectGetMaxY([UIApplication sharedApplication].statusBarFrame);
    flexWindow.frame = CGRectMake(0.5 * ([UIScreen mainScreen].bounds.size.width - 30), windowY, 30, 13);
    [flexWindow makeKeyAndVisible];
    
    UIButton *flexBtn = [UIButton buttonWithType:UIButtonTypeCustom];
    flexBtn.titleLabel.font = [UIFont systemFontOfSize:12];
    [flexBtn setTitle:@"FLEX" forState:UIControlStateNormal];
    [flexBtn setTitleColor:[UIColor blueColor] forState:UIControlStateNormal];
    [flexBtn addTarget:self action:@selector(showFlexSettingView) forControlEvents:UIControlEventTouchUpInside];
    flexBtn.frame = flexWindow.bounds;
    [flexWindow addSubview:flexBtn];
    
    flexWindow.hidden = [[[IRCacheManager sharedInstance] objectForKey:kFlexDebugDisableCacheKey] boolValue];
}

-(void)showFlexSettingView
{
    [[FLEXManager sharedManager] showExplorer];
}

@end

File: iRead/iRead/App/AppDelegate+Debug.swift
//
//  AppDelegate+Debug.swift
//  iRead
//
//  Created by zzyong on 2021/12/5.
//  Copyright © 2021 iread.com. All rights reserved.
//

#if DEBUG

import FLEX
import UIKit

public let flexWindow: UIWindow = UIWindow()

extension AppDelegate {
    
    func setupDebugConfig() {
        addFlexDebugView()
    }
    
    func addFlexDebugView() {
        
        flexWindow.backgroundColor = .clear
        flexWindow.rootViewController = .init()
        flexWindow.windowLevel = UIWindow.Level.statusBar + 50;
        
        let flexY = UIApplication.shared.statusBarFrame.maxY
        let flexW: CGFloat = 30
        let flexX = (UIScreen.main.bounds.width - flexW) * 0.5
        flexWindow.frame = CGRect.init(x: flexX, y: flexY, width: flexW, height: 13)
        
        flexWindow.makeKeyAndVisible()
        
        let flexBtn: UIButton = .init(type: UIButton.ButtonType.custom)
        flexBtn.titleLabel?.font = .systemFont(ofSize: 12);
        flexBtn.setTitle("FLEX", for: UIControl.State.normal)
        flexBtn.setTitleColor(.blue, for: UIControl.State.normal)
        flexBtn.addTarget(self, action:#selector(AppDelegate.showFlexDebugView), for: UIControl.Event.touchUpInside)
        flexBtn.frame = flexWindow.bounds
        flexWindow.addSubview(flexBtn)
        flexWindow.isHidden = !UserDefaults.standard.bool(forKey: kEnableFlex)
    }
    
    @objc func showFlexDebugView() {
        FLEXManager.shared.showExplorer()
    }
}

#endif
